ostd/src/timer/jiffies.rs

```diff
@@ -32,14 +36,36 @@ impl Jiffies {
         self.0
     }
 
+    /// Adds the given number of jiffies, saturating at [`Jiffies::MAX`] on overflow.
+    pub fn add(&mut self, jiffies: u64) {
+        self.0 = self.0.saturating_add(jiffies);
+    }
+
     /// Gets the [`Duration`] calculated from the jiffies counts.
-    pub fn as_duration(self) -> Duration {
+    fn as_duration(self) -> Duration {
         Duration::from_millis(self.0 * 1000 / TIMER_FREQ)
     }
+
+    /// Gets the jiffies counts calculated from the [`Duration`].
+    ///
+    /// Returns a `Jiffies::MAX` on overflow.
+    pub fn from_duration(duration: Duration) -> Self {
+        let millis = duration.as_millis();
+        let jiffies = (millis * TIMER_FREQ as u128).div_ceil(1000);
+        jiffies.try_into().map(Self::new).unwrap_or(Self::MAX)
+    }
 }
 
 impl From<Jiffies> for Duration {
     fn from(value: Jiffies) -> Self {
         value.as_duration()
     }
 }
+
+impl Add for Jiffies {
```

Comment:
What does `Jiffies::elapsed() + Jiffies::elapsed()` suppose to mean? It seems mean nothing.

If `Jiffies` is more like an [`Instant`](https://doc.rust-lang.org/std/time/struct.Instant.html) (which also has an [`elapsed`](https://doc.rust-lang.org/std/time/struct.Instant.html#method.elapsed) method), perhaps it shouldn't implement `Add`.